src/ISystemStaking.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8;
+
+struct BucketInfo {
+    uint256 typeIndex;
+    uint256 unlockedAt; // UINT256_MAX: in lock
+    uint256 unstakedAt; // UINT256_MAX: in stake
+    address delegate;
+}
+
+struct BucketType {
+    uint256 amount;
+    uint256 duration;
+    uint256 activatedAt;
+}
+
+interface ISystemStaking {
+
+    /* Pausable interface */
+    
+    event Paused(address account);
+    event Unpaused(address account);
+    function paused() external view returns (bool);
+    function _pause() external;
+    function _unpause() external;
+
+    /* Ownable interface */
+
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+    function owner() external view returns (address);
+    function renounceOwnership() external;
+    function transferOwnership(address newOwner) external;
+
+    /* ERC721 interface */
```

Comment:
missing three functions in ERC721:
- function name() external view returns (string memory);
- function symbol() external view returns (string memory);
- function tokenURI(uint256 tokenId) external view returns (string memory);